File: back/src/main/java/com/skytrix/model/dto/card/CardImageDTO.java
package com.skytrix.model.dto.card;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class CardImageDTO {
    private Long id;
    private Long imageId;
    private String url;
    private String smallUrl;
    private Long cardId;
}

File: front/src/app/core/model/dto/card-image-dto.ts
export class CardImageDTO {
  id: number;
  imageId: number;
  url: string;
  smallUrl: string;
  cardId: number;

  constructor(cardImageDTO?: CardImageDTO) {
    this.id = cardImageDTO?.id ?? 0;
    this.imageId = cardImageDTO?.imageId ?? 0;
    this.url = cardImageDTO?.url ?? '';
    this.smallUrl = cardImageDTO?.smallUrl ?? '';
    this.cardId = cardImageDTO?.cardId ?? 0;
  }
}

export class IndexedCardImageDTO {
  id: number;
  index: number;
  image: CardImageDTO;

  constructor(IndexedCardImageDTO: IndexedCardImageDTO) {
    this.id = IndexedCardImageDTO.id;
    this.index = IndexedCardImageDTO.index;
    this.image = IndexedCardImageDTO.image;
  }
}
